src/Storages/TTLDescription.cpp

```diff
@@ -182,73 +184,41 @@ TTLDescription TTLDescription::getTTLFromAST(
             if (ttl_element->group_by_key.size() > pk_columns.size())
                 throw Exception("TTL Expression GROUP BY key should be a prefix of primary key", ErrorCodes::BAD_TTL_EXPRESSION);
 
-            NameSet primary_key_columns_set(pk_columns.begin(), pk_columns.end());
             NameSet aggregation_columns_set;
-
-            for (const auto & column : primary_key.expression->getRequiredColumns())
-                primary_key_columns_set.insert(column);
+            NameSet used_primary_key_columns_set;
 
             for (size_t i = 0; i < ttl_element->group_by_key.size(); ++i)
             {
                 if (ttl_element->group_by_key[i]->getColumnName() != pk_columns[i])
                     throw Exception(
                         "TTL Expression GROUP BY key should be a prefix of primary key",
                         ErrorCodes::BAD_TTL_EXPRESSION);
-            }
 
-            for (const auto & [name, value] : ttl_element->group_by_aggregations)
-            {
-                if (primary_key_columns_set.count(name))
-                    throw Exception(
-                        "Can not set custom aggregation for column in primary key in TTL Expression",
-                        ErrorCodes::BAD_TTL_EXPRESSION);
+                used_primary_key_columns_set.insert(pk_columns[i]);
+            }
 
+            for (const auto & [name, _] : ttl_element->group_by_aggregations)
                 aggregation_columns_set.insert(name);
-            }
 
             if (aggregation_columns_set.size() != ttl_element->group_by_aggregations.size())
                 throw Exception(
                     "Multiple aggregations set for one column in TTL Expression",
                     ErrorCodes::BAD_TTL_EXPRESSION);
 
-
             result.group_by_keys = Names(pk_columns.begin(), pk_columns.begin() + ttl_element->group_by_key.size());
-
             auto aggregations = ttl_element->group_by_aggregations;
 
-            for (size_t i = 0; i < pk_columns.size(); ++i)
+            const auto & primary_key_expressions = primary_key.expression_list_ast->children;
+            for (size_t i = ttl_element->group_by_key.size(); i < primary_key_expressions.size(); ++i)
```

Comment:
Here we just add all unused primary key columns with `any` aggregation function? Need a comment.